blueocean-dashboard/src/main/js/creation/bitbucket/server/steps/BbChooseServerStep.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { PropTypes } from 'react';
+import { observer } from 'mobx-react';
+import { Dropdown, FormElement } from '@jenkins-cd/design-language';
+
+import FlowStep from '../../../flow2/FlowStep';
+import BbAddServerDialog from '../BbAddServerDialog';
+
+
+let t = null;
+
+
+@observer
+class BbChooseServerStep extends React.Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedServer: null,
+            showAddServerDialog: false,
+            urlErrorMsg: null,
+        };
+
+        this.dropdown = null;
+    }
+
+    componentWillMount() {
+        t = this.props.flowManager.translate;
+    }
+
+    _onChangeServerDropdown(option) {
+        const { serverManager } = this.props.flowManager;
+        serverManager.validateVersion(option.id)
```

Comment:
This is fine. This is to take care of cases where user adds server via classic UI, so you could still have servers that are not of supported version.

When user adds new servers, we check the version in backend, see https://github.com/jenkinsci/blueocean-plugin/blob/bbc72facebc13eaa320d85b0f04f23e6b019a240/blueocean-bitbucket-pipeline/src/main/java/io/jenkins/blueocean/blueocean_bitbucket_pipeline/server/BitbucketServerEndpointContainer.java#L59-L59.